kernel/rtlil.cc

```diff
@@ -639,7 +665,18 @@ RTLIL::Module::~Module()
 		delete it->second;
 	for (auto it = processes.begin(); it != processes.end(); ++it)
 		delete it->second;
+#ifdef WITH_PYTHON
+	RTLIL::Module::get_all_modules()->erase(hashidx_);
+#endif
+}
+
+#ifdef WITH_PYTHON
+static std::map<unsigned int, RTLIL::Module*> *all_modules = new std::map<unsigned int, RTLIL::Module*>();
```

Comment:
Is there any reason why this is a pointer allocated by calling new at startup, rather than a simple static object?

Reply:
No :) 827a96d3a38afe025c9efbd182069a9c9adee267